SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp

```diff
@@ -115,17 +115,36 @@ void ComponentReader::ApplyComponentData(const Worker_ComponentId ComponentId, S
 
 	Schema_Object* ComponentObject = Schema_GetComponentDataFields(Data);
 
-	TArray<uint32> UpdatedIds;
-	UpdatedIds.SetNumUninitialized(Schema_GetUniqueFieldIdCount(ComponentObject));
-	Schema_GetUniqueFieldIds(ComponentObject, UpdatedIds.GetData());
+	// ComponentData will be missing fields if they are completely empty (options, lists, and maps).
+	// However, we still want to apply this empty data, so we need to reconstruct the full
+	// list of field IDs for that component type (Data, OwnerOnly).
+	const TArray<Schema_FieldId>& InitialIds = ClassInfoManager->GetFieldIdsByComponentId(ComponentId);
+#if DO_GUARD_SLOW
```

Comment:
I think there was some conversation about this before - but what value is this check adding? Is it just checking random things in the hope that more stuff will catch errors in how we replicate? Or are we using this to try and catch specific issues around using the mismatched binaries and schema databases? Although this would only catch that in specific cases where the two had different numbers of fields.

If we really want this check it would also be better to encapsulate this in a function if possible. As this is most of the code in this function and it has little to do with it's actual work.

Reply:
Side note - as these are two ordered lists we can do this in linear rather than quadratic time by walking through them in step. I'm guessing unreal already has set based functions to do this.

Reply:
I guess, yeah, currently the only thing this would catch are mismatched binaries and schema dbs. The initial motivation for including this is due to not being confident that the fix is correct. I am more confident that this is correct now, so it could be argued that this can be removed.

Reply:
I have decided to remove the slow guard.